src/ResourceManager/ContainerRegistry/Commands.ContainerRegistry/Commands/NewAzureContainerRegistry.cs

```diff
@@ -0,0 +1,106 @@
+{
+    [Cmdlet(VerbsCommon.New, ContainerRegistryNoun), OutputType(typeof(PSContainerRegistry))]
+    public class NewAzureContainerRegistry : ContainerRegistryCmdletBase
+    {
+        [Parameter(
+            Position = 0,
+            Mandatory = true,
+            ValueFromPipelineByPropertyName = true,
+            HelpMessage = "Resource Group Name.")]
+        [ValidateNotNullOrEmpty]
+        public string ResourceGroupName { get; set; }
+
+        [Parameter(
+            Position = 1,
+            Mandatory = true,
+            ValueFromPipelineByPropertyName = true,
+            HelpMessage = "Container Registry Name.")]
+        [Alias(ContainerRegistryNameAlias, RegistryNameAlias, ResourceNameAlias)]
+        [ValidateNotNullOrEmpty]
+        public string Name { get; set; }
+
+        [Parameter(
+            Position = 2,
+            Mandatory = true,
+            ValueFromPipelineByPropertyName = true,
+            HelpMessage = "Container Registry SKU. " + AllowedSkuNames)]
+        [Alias(ContainerRegistrySkuAlias, RegistrySkuAlias)]
+        [ValidateNotNullOrEmpty]
+        [ValidateSet(SkuTier.Basic, IgnoreCase = false)]
+        public string Sku { get; set; }
+
+        [Parameter(
+            Mandatory = false,
+            HelpMessage = "Container Registry Location. Default to the location of the resource group.")]
+        [ValidateNotNullOrEmpty]
+        public string Location { get; set; }
+
+        [Parameter(
+            Mandatory = false,
+            HelpMessage = "Indicates whether the admin user is enabled.")]
+        [ValidateNotNull]
+        [Alias(AdminEnabledAlias)]
+        public bool? AdminUserEnabled { get; set; }
+
+        [Parameter(
+            Mandatory = false,
+            HelpMessage = "Container Registry Tags.")]
+        [ValidateNotNull]
+        [Alias(TagsAlias)]
+        public Hashtable Tag { get; set; }
+
+        [Parameter(
+            Mandatory = false,
+            HelpMessage = "The name of an existing storage account.")]
+        public string StorageAccountName { get; set; }
+
+        public override void ExecuteCmdlet()
+        {
+            var registryNameStatus = RegistryClient.CheckRegistryNameAvailability(Name);
+
+            if (registryNameStatus?.NameAvailable != null && !registryNameStatus.NameAvailable.Value)
+            {
+                throw new InvalidOperationException(registryNameStatus.Message);
+            }
+
+            var tags = TagsConversionHelper.CreateTagDictionary(Tag, validate: true);
+
+            if (Location == null)
+            {
+                Location = ResourceManagerClient.GetResourceGroupLocation(ResourceGroupName);
+            }
+
+            DeploymentExtended result = ResourceManagerClient.CreateRegistry(
+                ResourceGroupName, Name, Location, Sku, AdminUserEnabled, StorageAccountName, tags);
+
+            if (result.Properties.ProvisioningState == DeploymentState.Succeeded.ToString())
```

Comment:
The polling logic of template deployment will print the error message if it is unsuccessful. This if block here is just another GET to print the created registry object if it is successful.